### ISSUE TITLE
[question] traineval mode in tf2/train.py

### ISSUE THREAD

**Issue by danielTobon43** [google/automl#1094]
Hi there,

I would like to know if the continuos eval section in the train.py is working since I noticed that, that block of code is never executed because the FLAGS.mode is always pointing to "train".

```
if 'train' in FLAGS.mode:
      val_dataset = get_dataset(False, config) if 'eval' in FLAGS.mode else None
      model.fit(
          get_dataset(True, config),
          epochs=config.num_epochs,
          steps_per_epoch=steps_per_epoch,
          callbacks=train_lib.get_callbacks(config.as_dict(), val_dataset),
          validation_data=val_dataset,
          validation_steps=(FLAGS.eval_samples // FLAGS.batch_size),
          verbose=2)
    else:
      # Continuous eval.
      for ckpt in tf.train.checkpoints_iterator(
              FLAGS.model_dir, min_interval_secs=180):
        logging.info('Starting to evaluate.')
        # Terminate eval job when final checkpoint is reached.
        try:
```
If I set `mode=train` then is executed the first block, if I use `mode=traineval` keeps executing the first block. So I changed the `if 'train' in FLAGS.mode:` to `if 'train' == FLAGS.mode:` and now the second block is executed, but if that happens then, the training process is never executed. Could someone helpme how is the process to eval the model performance? I need to get the AP scores in the same way with the version for TF1 but with the version TF2 of the efficientdet.

Thanks

**Comment by danielTobon43** [comment]
For anyone trying to evaluate the model I changed the second block:

```
if 'train' == FLAGS.mode:
            val_dataset = get_dataset(False, config) if 'eval' in FLAGS.mode else None
            model.fit(
                get_dataset(True, config),
                epochs=config.num_epochs,
                steps_per_epoch=steps_per_epoch,
                callbacks=train_lib.get_callbacks(config.as_dict(), val_dataset=val_dataset),
                validation_data=val_dataset,
                validation_steps=(FLAGS.eval_samples // FLAGS.batch_size),
                verbose=2)
        else:
            val_dataset = get_dataset(False, config) if 'eval' in FLAGS.mode else None
            custom_call = train_lib.get_callbacks(config.as_dict(), val_dataset=val_dataset)
            coco_eval = train_lib.COCOCallback(val_dataset, 1)
            custom_call.append(coco_eval)
            model.fit(
                get_dataset(True, config),
                epochs=config.num_epochs,
                steps_per_epoch=steps_per_epoch,
                callbacks=custom_call,
                validation_data=val_dataset,
                validation_steps=(FLAGS.eval_samples // FLAGS.batch_size),
                verbose=2)
```
And now I am getting the AP metric in the coco folder for tensorboard
